src/parser/transform/expression/transform_array_access.cpp

```diff
@@ -6,6 +6,24 @@
 
 namespace duckdb {
 
+void Transformer::DetermineSliceBounds(duckdb_libpgquery::PGNode *idx, unique_ptr<ParsedExpression> &bound,
+                                       int64_t limit) {
+	if (idx) {
+		bound = TransformExpression(idx);
```

Comment:
Is this special casing necessary? Can't we default to `-2^63` and `2^63` as lower/upper bound and also allow them to be input as user input? Will anything produce an incorrect result?

Reply:
So far I have found the only case that produces an incorrect result is when the step is negative because it swaps the upper and lower bound. 
In python:
```python
>>> [1,2,3,4,5][::-1]
[5, 4, 3, 2, 1]
>>> [1,2,3,4,5][-9223372036854775808:9223372036854775807:-1]
[]
```

But because we default to min/max numeric limit when the bounds are left empty, if we don't differentiate between user input and default, both queries will either always produce the first result or the second result (depending on how we handle it), so this is the best solution we could come up with, but I'm definitely open to ideas as I realize this isn't ideal.

Reply:
Adding a new expression seems too heavy-handed, can we use e.g. an empty constant list or so (e.g. `Value::LIST(LogicalType::INTEGER, vector<Value>());`?

Reply:
I've implemented this and it actually works a lot better and also prevents a lot of the issues we were having with the numeric limit! :)